AutomatedTesting/Gem/PythonTests/editor_test_testing/TestSuite_Main.py

```diff
@@ -3,28 +3,21 @@
 For complete copyright and license terms please see the LICENSE at the root of this distribution.
 
 SPDX-License-Identifier: Apache-2.0 OR MIT
-"""
 
-"""
 This suite contains the tests for editor_test utilities.
 """
-
 import pytest
 import os
 import sys
 import importlib
-import re
 
 import ly_test_tools
-from ly_test_tools import LAUNCHERS
-
-sys.path.append(os.path.dirname(os.path.abspath(__file__)))
-
+import ly_test_tools.environment.process_utils as process_utils
 from ly_test_tools.o3de.editor_test import EditorSingleTest, EditorSharedTest, EditorTestSuite, Result
 from ly_test_tools.o3de.asset_processor import AssetProcessor
-import ly_test_tools.environment.process_utils as process_utils
 
-import argparse, sys
+sys.path.append(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
Is this line even needed here? Its only purpose seems to be a minor simplifying of relative imports below similar to `import EditorTest_That_Passes` ...can those imports instead be written correctly?

If it is necessary it's better to do it after the other imports, but I still don't like to see this type of classpath hack that could interfere with future imported modules, including other tests. Luckily it would currently be limited to tests within the same CTest invocation.

Reply:
The AP is being kept open between tests, so the only thing would be Editor startup/teardown (which this test needs).

The sys.path line is unfortunately needed for this. I think maybe we can add test modules' packages to the PYTHON_PATH to remove the need for this, but that may be out of scope from this.

Reply:
Ah, I see now that this test-code is micromanaging the AP in a way that other tests do not. I'll still be writing a request to automatically deduplicate this in a similar way for EditorTest.

Ah, looks like the issue comes from loading a script/main instead of things living inside modules: https://stackoverflow.com/questions/14132789/relative-imports-for-the-billionth-time

Modifying PYTHONPATH and modfiying sys.path is essentially achieving the same thing. It'd be a question on whether something like EditorTest will know enough to pre-include the correct path, which I don't think it will. Plus it could be surprising to only have some directories available in that way. Blah.